core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraFromRjsAdapter.kt

```diff
@@ -49,63 +68,498 @@ import fr.sncf.osrd.utils.distanceRangeMapOf
+        val curveGradient = if (currentCurve.value == 0.0) 0.0 else 800.0 / abs(currentCurve.value)
+        val upper = minOf(currentSlope.upper, currentCurve.upper)
+        gradients.put(lower, upper, curveGradient + slopeGradient)
+        if ((currentSlope.upper == upper) and currentSlopeIt.hasNext()) {
+            currentSlope = currentSlopeIt.next()
+        }
+        if ((currentCurve.upper == upper) and currentCurveIt.hasNext()) {
+            currentCurve = currentCurveIt.next()
+        }
+        lower = upper
+    }
+
+    return gradients
+}
+
+/** Builds the ranges of blocked loading gauge types on the track */
+private fun getBlockedGauge(rjsSection: RJSTrackSection): DistanceRangeMap<LoadingGaugeConstraint> {
+    // This method has a bad complexity compared to more advanced solutions,
+    // but we don't expect more than a few ranges per section.
+    // TODO: use an interval tree
+    val res = distanceRangeMapOf<LoadingGaugeConstraint>()
+    if ((rjsSection.loadingGaugeLimits == null) || (rjsSection.loadingGaugeLimits.isEmpty()))
+        return res
+
+    // Sorts and removes duplicates
+    val transitions = TreeSet<Double>()
+    for (range in rjsSection.loadingGaugeLimits) {
+        transitions.add(range.begin)
+        transitions.add(range.end)
+    }
+
+    val transitionsList = ArrayList(transitions) // Needed for index based loop
+    for (i in 1 until transitionsList.size) {
+        val begin = transitionsList[i - 1]
+        val end = transitionsList[i]
+        val allowedTypes = HashSet<RJSLoadingGaugeType>()
+        for (range in rjsSection.loadingGaugeLimits) {
+            if (range.begin <= begin && range.end >= end) {
+                val compatibleTypes = compatibleGaugeTypeMap[range.category]
+                if (compatibleTypes == null) {
+                    logger.warn("Invalid gauge type ${range.category} for track ${rjsSection.id}")
+                    allowedTypes.addAll(allLoadingGaugeTypeSet)
+                } else {
+                    allowedTypes.addAll(compatibleTypes)
+                }
+            }
+        }
+        val blockedTypes = Sets.difference(enumValues<RJSLoadingGaugeType>().toSet(), allowedTypes)
+        res.put(
+            begin.meters,
+            end.meters,
+            LoadingGaugeConstraintImpl(ImmutableSet.copyOf(blockedTypes))
+        )
+    }
+    return res
+}
+
+private fun buildCompatibleGaugeTypesMap(): Map<RJSLoadingGaugeType, Set<RJSLoadingGaugeType>> {
+    val compatibleGaugeType = hashMapOf<RJSLoadingGaugeType, Set<RJSLoadingGaugeType>>()
+    for (gaugeType in enumValues<RJSLoadingGaugeType>()) {
+        val compatibleTypes = gaugeType.compatibleGaugeTypes
+        if (compatibleTypes != null) compatibleGaugeType[gaugeType] = compatibleTypes
+    }
+    return compatibleGaugeType
+}
+
+@JvmField val compatibleGaugeTypeMap = buildCompatibleGaugeTypesMap()
+@JvmField val allLoadingGaugeTypeSet = enumValues<RJSLoadingGaugeType>().toSet()
```

Comment:
```suggestion
val compatibleGaugeTypeMap = buildCompatibleGaugeTypesMap()
val allLoadingGaugeTypeSet = enumValues<RJSLoadingGaugeType>().toSet()
```

I don't think it's required when used only from kotlin files

Reply:
Done in https://github.com/osrd-project/osrd/pull/6722/commits/d47d21a0d5092e3184dfafbb79bc63d946fb12ed